src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -176,92 +162,100 @@ def __init__(
         # Set gains
         self.set_channel_gains(gains)
 
+
         # Set offsets
-        offset = self._es.offset if hasattr(self._es, "offset") else 0
-        if offset == 0 and "offset" in self._nwbfile.electrodes:
-            electrode_table_index = self._es.electrodes.data[:]
-            offset = self._nwbfile.electrodes["offset"].data[electrode_table_index]
+        offset = electrical_series.offset if hasattr(electrical_series, "offset") else 0
+        if offset == 0 and "offset" in electrodes_table:
+            offset = electrodes_table["offset"].data[electrodes_indices]
 
         self.set_channel_offsets(offset * 1e6)
 
-        # Add properties
+        #########
+        # Extract and re-name properties from nwbfile TODO: Should be a function
+        ########
+
         properties = dict()
-        for es_ind, (channel_id, electrode_table_index) in enumerate(zip(channel_ids, self._es.electrodes.data)):
-            if "rel_x" in self._nwbfile.electrodes:
-                ndim = 2  # assume 2 dimensions
-                if "rel_z" in self._nwbfile.electrodes:
-                    ndim = 3  # if we have rel_z, it is 3 dimensions
-
-                if "location" not in properties:
-                    properties["location"] = np.zeros((self.get_num_channels(), ndim), dtype=float)
-                properties["location"][es_ind, 0] = self._nwbfile.electrodes["rel_x"][electrode_table_index]
-                if "rel_y" in self._nwbfile.electrodes:
-                    properties["location"][es_ind, 1] = self._nwbfile.electrodes["rel_y"][electrode_table_index]
-                if "rel_z" in self._nwbfile.electrodes:
-                    properties["location"][es_ind, 2] = self._nwbfile.electrodes["rel_z"][electrode_table_index]
-
-            for col in self._nwbfile.electrodes.colnames:
-                if isinstance(self._nwbfile.electrodes[col][electrode_table_index], ElectrodeGroup):
+        # Extract rel_x, rel_y and rel_z and assign to location
+
+        # TODO: Refactor ALL of this and add tests. This is difficult to read.
+        if "rel_x" in electrodes_table:
+            ndim = 3 if "rel_z" in electrodes_table else 2
+            properties["location"] = np.zeros((self.get_num_channels(), ndim), dtype=float)
+
+        for es_ind, (channel_id, electrode_table_index) in enumerate(zip(channel_ids, electrodes_indices)):
+            if "rel_x" in electrodes_table:
+                properties["location"][es_ind, 0] = electrodes_table["rel_x"][electrode_table_index]
+                if "rel_y" in electrodes_table:
+                    properties["location"][es_ind, 1] = electrodes_table["rel_y"][electrode_table_index]
+                if "rel_z" in electrodes_table:
+                    properties["location"][es_ind, 2] = electrodes_table["rel_z"][electrode_table_index]
+
+        # Extract all the other properties
+        for es_ind, (channel_id, electrode_table_index) in enumerate(zip(channel_ids, electrodes_indices)):
```

Comment:
Since style and readability seems to be the main purpose of the PR, I'd suggest also renaming `es_ind` to `electrical_series_index`